pkg/policy/l4.go

```diff
@@ -729,9 +729,13 @@ func createL4Filter(policyCtx PolicyContext, peerEndpoints api.EndpointSelectorS
 
 		// Override the parser type to CRD is applicable.
 		forceRedirect := false
-		if pr.Listener != "" {
+		if pr.Listener != nil {
 			l4.L7Parser = ParserTypeCRD
-			l4.Listener = pr.Listener
+			ns := pr.Listener.Namespace
+			if ns == "" {
```

Comment:
Should probably check listener namespace matches policy namespace (for non-CCNP's)